tx-backend/src/main/java/org/eclipse/tractusx/traceability/common/model/SecurityUtils.java

```diff
@@ -20,21 +20,69 @@
 package org.eclipse.tractusx.traceability.common.model;
 
 
+import org.eclipse.tractusx.traceability.qualitynotification.domain.base.model.QualityNotificationAffectedPart;
+import org.eclipse.tractusx.traceability.qualitynotification.infrastructure.edc.model.EDCNotification;
+import org.eclipse.tractusx.traceability.qualitynotification.infrastructure.edc.model.EDCNotificationContent;
+import org.eclipse.tractusx.traceability.qualitynotification.infrastructure.edc.model.EDCNotificationHeader;
+
 import java.util.ArrayList;
 import java.util.List;
+import java.util.Objects;
 
 public class SecurityUtils {
 
     public static String sanitize(String unSanitizedInput) {
-        return unSanitizedInput.replaceAll("\r\n|\r|\n", " ");
+        if (unSanitizedInput != null) {
+            return unSanitizedInput.replaceAll("\r\n|\r|\n", " ");
+        }
+        return null;
     }
 
     public static List<String> sanitize(List<String> unSanitizedList) {
-        List<String> cleanListOfAffectedItems = new ArrayList<>();
-        for (String affectedItem : unSanitizedList) {
-            String cleanAffectedItem = sanitize(affectedItem);
-            cleanListOfAffectedItems.add(cleanAffectedItem);
+        if (unSanitizedList != null) {
+            List<String> cleanListOfAffectedItems = new ArrayList<>();
+            for (String affectedItem : unSanitizedList) {
+                String cleanAffectedItem = sanitize(affectedItem);
+                cleanListOfAffectedItems.add(cleanAffectedItem);
+            }
+            return cleanListOfAffectedItems;
+        }
+        return null;
+    }
+
+
+    public static EDCNotification sanitizeEDCNotification(EDCNotification edcNotification) {
```

Comment:
```suggestion
    public static EDCNotification sanitize(EDCNotification edcNotification) {
```
The parameter and return type already gives a clue about what is going to be sanitized